### DISCUSSION TITLE
Hooks scopes

### DISCUSSION THREAD

**Discussion by bdvorianov** [Escape-Technologies/mookme#79] [Q&A]
Imagine we have a structure like:

+ rootFolder
  + .hooks
  + package
    + .hooks
      + pre-commit.json
      
In such case, pre-commit will be fired only if `git commit` was fired from the `package` directory, but it seems kinda natural that it may be fired when `commit` was fired from the `rootFolder` and we have staged changes in `package`. I see global hook with `onlyOn` as a solution, but it's not as clean as if it would work as above. Is there a way to do this? 🤔

**Comment by LMaxence** [comment]
Hello ! 

The way ran hooks are selected is not based on where you fire the `git command`, but rather based on what you files are included in your commits. Once again, [this is all mentioned in the documentation](https://mookme.org/get-started/#configuration), but maybe it lacks some clarity, in which case feel free to open an issue so that we can make it clearer :) 

In your case, the global hooks (defined in the `.hooks` folder in your `rootFolder`) are always executed. Then, for a `.hooks` folder to be included in a run, you need to have commited files coming from the directory holding this `.root` folder.

It is done, regardless of where you `git commit` command is fired from.

**Reply by bdvorianov** [reply]
Thanks for your reply. 🙏🏻

Somewhy it doesn't work so for pre-push in my case. Pre-commit works as described, but not pre-push. So when I push from `rootFolder ` it doesn't work, but when I change directory to `package` and push from there it works as expected. I'm not sure, but maybe it's issue specific to `pre-push`.

**Reply by LMaxence** [reply]
hmmm that is definitely not normal, could you open an issue please ? Eventually with a minimal reproducible repository example ?

The pre-push resolution strategy does not use the files about to be commited but the files that have been commited since the last push

**Reply by bdvorianov** [reply]
I'll do it a bit later 👌🏻

**Reply by mhozhynewscorp** [reply]
hey guys @bdvorianov @LMaxence 
I created the sample repo and provided link in mentioned issue
Please take a look

**Reply by LMaxence** [reply]
Hello ! I'll try to find time to look around this this weekend :) Thanks for your feedback
